magento/magento2
# add sourceMap generation for less files in developer mode

### Description (*)

Currently when running magento 2 in developer mode with server-side less compilation it's very hard to detect where a style is coming from. The only workaround is to use Grunt. 
However the underlying library for less compilation supports sourcemap generation with a flag that can be toggled based on the mode magento is in. 

### Related Pull Requests


### Fixed Issues (if relevant)


### Manual testing scenarios (*)

1. Switch magento 2 in developer mode 
2. Let it regenerate css from the less files
3. Check inspector to see the source file responsible for the code 

![image](https://github.com/user-attachments/assets/437750da-39b0-4645-8f92-614982e28cf2)
![image](https://github.com/user-attachments/assets/07c656c0-d742-490c-89e3-184ad7c72d1f)


### Questions or comments


### Contribution checklist (*)


### Resolved issues:
1. [x] resolves magento/magento2#38982: add sourceMap generation for less files in developer mode
